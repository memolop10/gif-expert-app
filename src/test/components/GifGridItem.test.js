import React from 'react'
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Test in <GifGridItem/>', () => {

    const title = 'Hola so el titulo'
    const url = 'https://localhost/Goku.jpg'
    const wrapper = shallow( <GifGridItem title={title} url={ url }/> )

    test('should render the component correctly', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('it should have a paragraph with the title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim()).toBe( title );

    })

    test('Must have an image equal to the url and alt of the props', () => {
        const img = wrapper.find('img');
        //console.log(img.props().src)
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })
    
    test('Should have animate__bounce', () => {
        
        const div = wrapper.find('div');
        //console.log(div.hasClass('animate__bounce'));
        //const className = div.prop('className');

        expect( div.hasClass('animate__bounce') ).toBe(true);
        //expect( className.includes('animate__bounce') ).toBe( true );
    })
    

    
});
