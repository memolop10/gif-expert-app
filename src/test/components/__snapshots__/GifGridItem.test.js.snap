// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem/> should render the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce animate__fast"
>
  <img
    alt="Hola so el titulo"
    src="https://localhost/Goku.jpg"
  />
  <p>
    Hola so el titulo
  </p>
</div>
`;
